registry_no_code_module_integration_test.go

```diff
@@ -0,0 +1,327 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tfe
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRegistryNoCodeModulesCreate(t *testing.T) {
+	skipUnlessBeta(t)
+	client := testClient(t)
+	ctx := context.Background()
+
+	orgTest, orgTestCleanup := createOrganization(t, client)
+	defer orgTestCleanup()
+
+	t.Run("with valid options", func(t *testing.T) {
+		t.Run("with follow-latest-version and enabled", func(t *testing.T) {
+			registryModuleTest, registryModuleTestCleanup := createRegistryModule(t, client, orgTest, PrivateRegistry)
+			defer registryModuleTestCleanup()
+
+			options := RegistryNoCodeModuleCreateOptions{
+				FollowLatestVersion: Bool(true),
+				Enabled:             Bool(true),
+				RegistryModule:      registryModuleTest,
+			}
+
+			noCodeModule, err := client.RegistryNoCodeModules.Create(ctx, orgTest.Name, options)
+			require.NoError(t, err)
+			assert.NotEmpty(t, noCodeModule.ID)
+			require.NotEmpty(t, noCodeModule.Organization)
+			require.NotEmpty(t, noCodeModule.RegistryModule)
+			assert.Equal(t, *options.FollowLatestVersion, noCodeModule.FollowLatestVersion)
+			assert.Equal(t, *options.Enabled, noCodeModule.Enabled)
+			assert.Equal(t, orgTest.Name, noCodeModule.Organization.Name)
+			assert.Equal(t, noCodeModule.RegistryModule.ID, noCodeModule.RegistryModule.ID)
+		})
+		t.Run("with version pin and enabled", func(t *testing.T) {
+			registryModuleTest, _ := createRegistryModule(t, client, orgTest, PrivateRegistry)
+
+			options := RegistryModuleCreateVersionOptions{
+				Version: String("1.2.3"),
+			}
+			rmv, err := client.RegistryModules.CreateVersion(ctx, RegistryModuleID{
+				Organization: orgTest.Name,
+				Name:         registryModuleTest.Name,
+				Provider:     registryModuleTest.Provider,
+			}, options)
+			require.NoError(t, err)
+			require.NotEmpty(t, rmv.Version)
+
+			ncOptions := RegistryNoCodeModuleCreateOptions{
+				VersionPin:     String("1.2.3"),
+				Enabled:        Bool(true),
+				RegistryModule: registryModuleTest,
+			}
+
+			noCodeModule, err := client.RegistryNoCodeModules.Create(ctx, orgTest.Name, ncOptions)
+			require.NoError(t, err)
+			assert.NotEmpty(t, noCodeModule.ID)
+			require.NotEmpty(t, noCodeModule.Organization)
+			require.NotEmpty(t, noCodeModule.RegistryModule)
+			assert.Equal(t, false, noCodeModule.FollowLatestVersion)
+			assert.Equal(t, *ncOptions.Enabled, noCodeModule.Enabled)
+			assert.Equal(t, *ncOptions.VersionPin, noCodeModule.VersionPin)
+			assert.Equal(t, orgTest.Name, noCodeModule.Organization.Name)
+			assert.Equal(t, noCodeModule.RegistryModule.ID, noCodeModule.RegistryModule.ID)
```

Comment:
replace assertion that compares noCodeModule.RegistryModule.ID to itself
```suggestion
		assert.Equal(t, registryModuleTest.ID, noCodeModule.RegistryModule.ID)
```